paddle/operators/concat_op.cc

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/operators/concat_op.h"
+#include <vector>
+
+namespace paddle {
+namespace operators {
+using framework::Tensor;
+
+class ConcatOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+ protected:
+  void InferShape(const framework::InferShapeContext &ctx) const override {
+    auto ins = ctx.MultiInput<framework::Tensor>("X");
+    auto *out = ctx.Output<framework::Tensor>("Out");
+    const int axis = static_cast<int>(ctx.op_.GetAttr<int>("axis"));
+    int N = ins.size();
+
+    PADDLE_ENFORCE_GT(N, 1, "Input tensors count should >= 1.");
+
+    auto dim_zero = ins[0]->dims();
+    auto dim_zero_size = dim_zero.size();
+    auto concat_dim = dim_zero;
+    for (int i = 1; i < N; i++) {
+      PADDLE_ENFORCE_EQ(dim_zero_size, ins[i]->dims().size(),
+                        "input tensors should have the same dim size.");
```

Comment:
the same rank
dim size is a little confused

Reply:
Done, we only need to keep the same dim except the axis.